File: Intersection-of-Two-Arrays.java
public class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> set1 = new HashSet<>();
        for (int num : nums1) {
            set1.add(num);
        }
        
        ArrayList<Integer> result = new ArrayList<>();
        for (int num : nums2) {
            if (set1.contains(num)) {
                result.add(num);
                set1.remove(num); 
            }
        }
        
        int[] res = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            res[i] = result.get(i);
        }
        
        return res;
    }
}

File: Intersection-of-Two-Arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let set1 = new Set(nums1);
    let result = [];
    for (let num of nums2) {
        if (set1.has(num)) {
            result.push(num);
            set1.delete(num); 
        }
    }
    return result;
};
